### PR TITLE
DHFPROD-6289:Add ESLint to CI/CD

### PR DESCRIPTION
**Pull request by yafeldman**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub-central/build.gradle [hunk 1]
```diff
@@ -194,6 +194,13 @@ task copyUiFiles(type: Copy, dependsOn: deleteSpringBootUiFiles, group: taskGrou
     into springBootUiPath
 }
 
+task lintUI(type: NpmTask, dependsOn: installDependencies, group: taskGroup) {
+    description = "linting the UI files"
```
Comment (rjrudin) [inline marklogic-data-hub-central/build.gradle:198]: Can you expand the description to define what happens when the task is run? For example, if no errors are found, is any report file written? If errors are found, what happens? Does the task fail, or does it print a report somewhere?

**Review by rjrudin — CHANGES_REQUESTED** [review]

File: marklogic-data-hub-central/build.gradle (hunk 1 above)
Comment (yafeldman) [reply marklogic-data-hub-central/build.gradle:198]: Before committing description here is the current implementation: 
    - if no errors, the success email is sent with message 'TestUnit passed'. 
    - If lint task is failed the job is terminated as failure and no notifications are sent.

**Review by rjrudin — APPROVED** [review]

**Review by SameeraPriyathamTadikonda — APPROVED** [review]

**Review by wooldridge — COMMENTED** [review]
@SameeraPriyathamTadikonda, the UI developers run the linting tests from the /ui directory, is this how the CI/CD tests will run?

**Comment by SameeraPriyathamTadikonda** [comment]
> @SameeraPriyathamTadikonda, the UI developers run the linting tests from the /ui directory, is this how the CI/CD tests will run?

Yes @wooldridge. We just created a gradle task to run linting. so that the node and npm versions are consistent with builds.

**Comment by wooldridge** [comment]
> > @SameeraPriyathamTadikonda, the UI developers run the linting tests from the /ui directory, is this how the CI/CD tests will run?
> 
> Yes @wooldridge. We just created a gradle task to run linting. so that the node and npm versions are consistent with builds.

Thanks for clarifying.

**Review by wooldridge — DISMISSED** [review]
@SameeraPriyathamTadikonda, the UI developers run the linting tests from the /ui directory, is this how the CI/CD tests will run?

File: Jenkinsfile [hunk 2]
```diff
@@ -287,7 +287,7 @@ void UnitTest(){
...
+        sh 'export JAVA_HOME=`eval echo "$JAVA_HOME_DIR"`;export GRADLE_USER_HOME=$WORKSPACE$GRADLE_DIR;export M2_HOME=$MAVEN_HOME/bin;export PATH=$JAVA_HOME/bin:$GRADLE_USER_HOME:$PATH:$MAVEN_HOME/bin;cd $WORKSPACE/data-hub;rm -rf $GRADLE_USER_HOME/caches;set +e;./gradlew clean;./gradlew marklogic-data-hub:bootstrap -i --stacktrace -PnodeDistributionBaseUrl=http://node-mirror.eng.marklogic.com:8080/;sleep 10s;./gradlew marklogic-data-hub-central:test -i --stacktrace -PnodeDistributionBaseUrl=http://node-mirror.eng.marklogic.com:8080/;sleep 10s;./gradlew marklogic-data-hub-central:lintUI -i --stacktrace -PnodeDistributionBaseUrl=http://node-mirror.eng.marklogic.com:8080/; |& tee console.log;sleep 10s;./gradlew ml-data-hub:test -i --stacktrace -PnodeDistributionBaseUrl=http://node-mirror.eng.marklogic.com:8080/;./gradlew web:test -i --stacktrace -PnodeDistributionBaseUrl=http://node-mirror.eng.marklogic.com:8080/;sleep 10s;./gradlew marklogic-data-hub-spark-connector:test -i --stacktrace -PnodeDistributionBaseUrl=http://node-mirror.eng.marklogic.com:8080/;'
```
Comment (SameeraPriyathamTadikonda) [inline Jenkinsfile:290]: Command wrong. 
it should be ./gradlew marklogic-data-hub-central:test -i --stacktrace -PnodeDistributionBaseUrl=http://node-mirror.eng.marklogic.com:8080/ |& tee console.log; sleep 10s; ./gradlew marklogic-data-hub-central:lintUI -i --stacktrace -PnodeDistributionBaseUrl=http://node-mirror.eng.marklogic.com:8080/;

**Review by SameeraPriyathamTadikonda — CHANGES_REQUESTED** [review]
Need changes

File: Jenkinsfile (hunk 2 above)
Comment (yafeldman) [reply Jenkinsfile]: Fixed

File: Jenkinsfile [hunk 3]
```diff
@@ -287,7 +287,7 @@ void UnitTest(){
...
+        sh 'export JAVA_HOME=`eval echo "$JAVA_HOME_DIR"`;export GRADLE_USER_HOME=$WORKSPACE$GRADLE_DIR;export M2_HOME=$MAVEN_HOME/bin;export PATH=$JAVA_HOME/bin:$GRADLE_USER_HOME:$PATH:$MAVEN_HOME/bin;cd $WORKSPACE/data-hub;rm -rf $GRADLE_USER_HOME/caches;set +e;./gradlew clean;./gradlew marklogic-data-hub:bootstrap -i --stacktrace -PnodeDistributionBaseUrl=http://node-mirror.eng.marklogic.com:8080/;sleep 10s;./gradlew marklogic-data-hub-central:test -i --stacktrace -PnodeDistributionBaseUrl=http://node-mirror.eng.marklogic.com:8080/;sleep 10s; |& tee console.log;sleep 10s;./gradlew ml-data-hub:test -i --stacktrace -PnodeDistributionBaseUrl=http://node-mirror.eng.marklogic.com:8080/;./gradlew web:test -i --stacktrace -PnodeDistributionBaseUrl=http://node-mirror.eng.marklogic.com:8080/;sleep 10s;./gradlew marklogic-data-hub-spark-connector:test -i --stacktrace -PnodeDistributionBaseUrl=http://node-mirror.eng.marklogic.com:8080/;./gradlew marklogic-data-hub-central:lintUI -i --stacktrace -PnodeDistributionBaseUrl=http://node-mirror.eng.marklogic.com:8080/;'
```
Comment (SameeraPriyathamTadikonda) [inline Jenkinsfile:290]: Please do not change the command. still has issues.

**Review by SameeraPriyathamTadikonda — CHANGES_REQUESTED** [review]

**Review by SameeraPriyathamTadikonda — APPROVED** [review]

**Review by wooldridge — APPROVED** [review]

**Pull request merged**